src/webserver/HttpBase.h

```diff
@@ -312,22 +312,21 @@ handle_request(
         return res;
     };
 
+    auto ticket = dosGuard.checkout(ip);
```

Comment:
Would you add a comment above `.checkout` that states that the resource is automatically checked in when `ticket` goes out of scope